src/uarch-state-access.h

```diff
@@ -127,6 +127,10 @@ class uarch_state_access : public i_uarch_state_access<uarch_state_access> {
         m_us.halt_flag = false;
     }
 
+    auto do_read_iflags() {
```

Comment:
Why use `auto` in function declaration? We usually are explicit about function return types, so I would change to `uint64_t`.